### ISSUE TITLE
EE recompiler performance improvement

### ISSUE THREAD

**Issue by andreynering** [TheRhysWyrill/XBSX2#51]
Seems that upstream got a considerable performance boost today:

- https://twitter.com/PCSX2/status/1593845946739146753
- https://github.com/PCSX2/pcsx2/pull/7295#issuecomment-1296031122

Do you plan to bring that patch into this fork?

**Comment by andreynering** [comment]
Just saw that some people mentioned on the Twitter thread that this patch supposedly contain changes to deliberately make it difficult to port to Xbox.

I don't know how much of that is true, but if accurate, that's certainly not a friendly attitude from them. My considerations.

You're free to proceed how you want, including closing this issue if you want. XBPS2 is fantastic as it is already, even without that patch.

I also enforce that you should consider GitHub Sponsors (or something else) so the community can support you somehow, if you want.

**Comment by refractionpcsx2** [comment]
It's some idiots taking what was said out of context.  If you look at the screenshot they are using for their abuse (because their users are going around calling us assholes and slagging us off for this), stenzek said the changes were anti-uwp in a sense that they won't work out of the box for UWP, if you read the *rest* of the screenshot, he says that UWP is doable if you understand the API.  There is nothing there which these developers cannot port over to UWP.

We aren't actively making it anti-UWP, we're writing code which works well on Windows/Linux and Mac, we have zero focus on XBox, so we aren't going to go out of our way to write favourable code for that, this is for the XBSX2 team to deal with, expecting us to write all their code for them is inherently ridiculous, if you're going to run a port, it's up to you to maintain it.

**Comment by danprice142** [comment]
Yeah but why do you guys support Aethersx2 fork and not this one?? After
all aethersx2 uses pcsx2 code but is closed source and offers donation
service while XBSX2 is open source and is not offered in the patron,
everyone in the Xbox Emulation Hub has tried to apologise to stenzek about
uploading the pcsx2 uwp early but it hasn’t helped in anyway and the Xbox
emulation hub or any member gets you guys stole code to sell from the pcsx2
community while you actively let aethersx2 distribute and offer a patrion,
makes no sense to me

**Comment by refractionpcsx2** [comment]
Aether didn't take Stenzeks code before it was finished.
They did not replace all the copyright headers
and they didn't recommit all our work without attribution.

This is why we have trouble. And not a single person has apologised and reversed their actions for using stenzeks early UWP work, the closes we got was the gamer13 guy pulling his upload from the store (as far as I know), it's still being used, and people are trying to justify the bad behaviour by saying "it was public!" it's still not cool.

Aether hasn't stolen anything, the source is open if you ask for it, the JIT's are closed, but that's their proprietary code and they are allowed to do that by the license, as long as they give the ability to recompile the emulator (which I believe they do), but everything has been done by the book, we made sure.

As I said on twitter, XBSX2 needs to right the wrongs they did for us to work together without problems, and those are.

1. Remove stenzeks UWP code and write their own.
2. Sync properly to master to give proper attribution.
3. restore the copyright header to their original form.

Further to this, they don't need us to proceed with this, as I have mentioned, they can do everything they need to do.  We do not need to do the code for them, nor do we do anything for Android directly, if there's anything android related in our code, it's because Tahlreth has upstreamed some of his changes, another thing XBSX2 has not done.

**Comment by stenzek** [comment]
This drama is getting ridiculous. I'm only posting here in the hope that things can be moved on. tldr; we tried to work things out, but there does not appear to be any interest in doing so, read the last paragraph.

> Yeah but why do you guys support Aethersx2 fork and not this one??

Emulators on the retail Windows store violates MS store policy [10.13.10](https://learn.microsoft.com/en-us/windows/uwp/publish/store-policies) `Products that emulate a game system or game platform are not allowed on any device family.`. I don't want anything I do to be associated with violating store policy. Developer mode is fine, that's what I originally intended my UWP port to be for. Ignoring all the other issues, considering this fork blatantly advertises the fact that it's on retail, legally to cover my arse, I personally want nothing to do with it, and I have no doubt the PCSX2 team members feel the same way.

> everyone in the Xbox Emulation Hub has tried to apologise to stenzek about uploading the pcsx2 uwp early but it hasn’t helped in anyway

Not one person has. Instead I got the opposite, mountains of abuse because I stopped working on it because I didn't want my work on the retail store, regardless of whether it was free/sold/behind a paywall. And there **were** instances of it being sold. Things were pretty quiet for the last couple of months, thankfully, until we merged my improvements into PCSX2, then all of a sudden there's been a slew of posts on various mediums calling me arrogant, asshole, a c**t, etc. Did I regret making the smartarse remark they're taking out of context? A little, sure, but the argument that PCSX2 is anti-UWP is flawed. As I stated on reddit, I have an up-to-date UWP branch which I used for benchmarking the EErec changes on an X1S, because it's the slowest CPU I have access to. There's nothing preventing XBSX2 from doing the same.

> uploading the pcsx2 uwp early

This is also potentially an issue. I'm not a lawyer, but as far as I understand, the license applies once you *distribute* the code. Since I never distributed the UWP port myself (as it was in development), I just made the mistake of pushing it to my GH repo, it may be argued that XBSX2 does not the right to use the UWP code, since some of those files did not include a copyright declaration/license header (see [GPL FAQ](https://www.gnu.org/licenses/gpl-faq.en.html#NoticeInSourceFile)). In the interest of everyone, I never pursued this, because I didn't want the stress of drama, but that's where the talk of "theft" comes from - it's akin to leaving the door to the house unlocked, someone walking in and taking valuables, there wasn't an exchange where the terms were outlined (distribution).

Refraction outlined the terms for cooperation above. The first one relates to the potential issue above. I also made a request here months ago (#27) to remove the non-GPL code and stop removing copyright, I believe the latter has been partially rectified, but the non-GPL code still remains in the repository. After that issue, https://github.com/TheRhysWyrill/XBSX2/commit/cc7d5cbb34675d1b7258590178cc0ade450c40dd#diff-1fc6c9d790720db6ba5a5ec8f1336281e4d653cac453403c478ebfc06e377038R396-R406 took code from DuckStation, without any attribution, and committed it under their name (the structure and variable names match 100%, and why would you include a completely redundant variable if it wasn't copied?). This happened **after** I reached out and tried to resolve things.

So, you're claiming that it's all the PCSX2 team's fault for "hurting the xbox community" - **we tried to work things out**. And _they just throw it back in our faces_. XBSX2 decided to fork, not us. We're not assholes for continuing to improve PCSX2, if you want to fork and get the latest improvements, then it's your responsibility to stay up to date. As refraction said, PCSX2 targets many platforms, there's no technical reason preventing fastmem and the other EErec improvements from working on UWP - I've done it as a proof of concept. If the "xbox community" weren't continuing to attempt to start drama and pointless fighting, and instead trying to work together and right the wrongs of the past, then perhaps I'd even be open to sharing patches.
